### PR TITLE
build!: upgrade to Node.js 20

### PR DESCRIPTION
**Pull request by tianfeng92**
Upgrade to node@v20 and bump related packages.

### CODE REVIEW

File: tests/integration/reporter.spec.js [hunk 1]
```diff
@@ -16,8 +16,8 @@ describe('runs tests on cloud', function () {
     expect(process.env.SAUCE_ACCESS_KEY).toBeDefined();
 
     const cucumberRunCommand = 'npx cucumber-js'
-    const format = path.join(process.cwd(), 'src/reporter.ts')
-    const args = `--config tests/integration/cucumber.json --format ${format} --publish-quiet`
+    const format = path.join(process.cwd(), 'src/reporter.js')
+    const args = `--config tests/integration/cucumber.json --format ${format}`
```
Comment (tianfeng92) [inline tests/integration/reporter.spec.js:20]: Cucumber updated module loading mechanism in 10.0.0
https://github.com/cucumber/cucumber-js/blob/main/UPGRADING.md#module-loading

`--publish-quiet` has been deprecated in 9.4.0 and will be removed in 11.0.0 or later. 

https://github.com/cucumber/cucumber-js/blob/842099b87b6d689c2a45f13c820fd382108346a8/docs/deprecations.md#publishquiet

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -24,6 +24,9 @@ jobs:
       - name: Lint
         run: npm run lint
 
+      - name: Npm Build
```
Comment (alexplischke) [inline .github/workflows/test.yml]: style: just for consistency
```suggestion
      - name: Build
```

**Comment by tianfeng92** [comment]
Hold a bit as I'm testing it on Windows. Something wrong for the integration test on windows and I'm verifying this part.

**Comment by tianfeng92** [comment]
> Hold a bit as I'm testing it on Windows. Something wrong for the integration test on windows and I'm verifying this part.

Nothing wrong with this reporter on Windows. Should be something weird within the runner.

File: tests/integration/reporter.spec.js [hunk 3]
```diff
@@ -16,8 +16,8 @@ describe('runs tests on cloud', function () {
     expect(process.env.SAUCE_ACCESS_KEY).toBeDefined();
 
     const cucumberRunCommand = 'npx cucumber-js'
-    const format = path.join(process.cwd(), 'src/reporter.ts')
```
Comment (alexplischke) [inline tests/integration/reporter.spec.js:19]: Why was this working before, but not anymore? 🤔

Comment (tianfeng92) [reply tests/integration/reporter.spec.js:19]: I'm also confused about this change.

File: src/reporter.ts [hunk 4]
```diff
@@ -91,6 +91,7 @@ export default class SauceReporter extends SummaryFormatter {
 
   logTestCase(testCaseFinished: { testCaseStartedId: string }) {
     const testCaseAttempt = this.eventDataCollector.getTestCaseAttempt(testCaseFinished.testCaseStartedId);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (alexplischke) [inline src/reporter.ts]: Why disable? Preferably, eslint continues to warn us.

Comment (alexplischke) [reply src/reporter.ts]: Just add a "warn" rule like so https://github.com/saucelabs/sauce-testcafe-runner/blob/71f68ccb12898d64153241689a4d562292f6c7d5/.eslintrc#L12-L12

Comment (tianfeng92) [reply src/reporter.ts]: Added.

File: .eslintrc.js [hunk 5]
```diff
@@ -19,6 +19,7 @@ module.exports = {
         "@typescript-eslint"
     ],
     rules: {
-        semi: "error"
+        semi: "error",
+        "@typescript-eslint/no-explicit-any": "off"
```
Comment (alexplischke) [inline .eslintrc.js]: We should still discourage the use, hence warning:
```suggestion
        "@typescript-eslint/no-explicit-any": "warn"
```

Comment (tianfeng92) [reply .eslintrc.js]: Oops. Misunderstood.

**Review by alexplischke — APPROVED** [review]

**Pull request merged**